content/en/blog/_posts/2022-09-21-csi-nodeexpandsecret.md

```diff
@@ -0,0 +1,205 @@
+---
+layout: blog
+title: >-
+  Kubernetes 1.25: Use Secrets for Node-Driven Expansion of CSI Volumes
+date: 2022-09-21
+slug: kubernetes-1-25-use-secrets-while-expanding-csi-volumes-on-node-alpha
+---
+
+**Author:** Humble Chirammal (Red Hat), Louis Koo (deeproute.ai)
+
+Kubernetes v1.25, released earlier this month, introduced a new feature
+that lets your cluster expand storage volumes, even when access to those
+volumes requires a secret (for example: a credential for accessing a SAN fabric)
+to perform node expand operation. This new behavior is in alpha and you
+must enable a feature gate (`CSINodeExpandSecret`) to make use of it.
+You must also be using [CSI](https://kubernetes-csi.github.io/docs/)
+storage; this change isn't relevant to storage drivers that are built in to Kubernetes.
+
+To turn on this new, alpha feature, you enable the `CSINodeExpandSecret` feature
+gate for the kube-apiserver and kubelet, which turns on a mechanism to send `secretRef`
+configuration as part of NodeExpansion  by the CSI drivers thus make use of
+the same to perform node side expansion operation with the underlying
+storage system.
+
+## What is this all about?
+
+Before Kubernetes v1.24, you were able to define a cluster-level StorageClass
+that made use of [StorageClass Secrets](https://kubernetes-csi.github.io/docs/secrets-and-credentials-storage-class.html),
+but you didn't have any mechanism to specify the credentials that would be used for
+operations that take place when the storage was mounted onto a node and when
+the volume has to be expanded at node side.
+
+The Kubernetes CSI already implemented a similar mechanism specific kinds of
+volume resizes; namely, resizes of PersistentVolumes where the resizes take place
+independently from any node referred as Controller Expansion. In that case, you
+associate a PersistentVolume with a Secret that contains credentials for volume resize
+actions, so that controller expansion can take place. CSI also supports a `nodeExpandVolume`
+operation which CSI drivers can make use independent of Controller Expansion or along with
+Controller Expansion on which, where the resize is driven from a node in your cluster where
+the volume is attached. Having resize (expansion) operations sometimes able to use credentials
+and sometimes not meant it was difficult for people to design CSI drivers that could cover all
+resize scenarios. Please refer [Volume Expansion GA blog]( VolumeExpansion GA blog https://kubernetes.io/blog/2022/05/05/volume-expansion-ga/)
```

Comment:
Can you please remove this sentence "Having resize (expansion) operations sometimes able to use credentials
and sometimes not meant it was difficult for people to design CSI drivers that could cover all
resize scenarios."

Reply:
sure, removed that @xing-yang . ptal. thanks.